### ISSUE TITLE
H12 timeout from heroku

### ISSUE THREAD

**Issue by jianqiaozhu** [Dallinger/Dallinger#5932]
Significant number of participants were rejected midway as they got timeout on heroku (error code H12).
I use the exact same experimental code but upgraded from Dallinger 9.8.0 to 9.10.0. The code deployed with Dallinger 9.8.10 worked well but not the 9.10.0. What could be the issue here?

**Comment by jacobyn** [comment]
It looks to me that there is another error that occurred at the same time, which H12 was related to. In Heroku you may want to check memory load in the "performance metrics" in Heroku GUI. One option is that your dyno was not big enough and you run out of memory.

**Comment by pmcharrison** [comment]
Yes, and check your logs to see what was the HTTP route that timed out.

**Comment by jianqiaozhu** [comment]
It is very likely that there are some problems with the way Dallinger assigns participant id. It turns out that Dallinger can assign identical participant_id to different participants with different worker_id as long as everybody agrees to consent and wait to start the experiment around the same time. I have replicated this error in the incognito mode. Any idea to avoid assigning same participant_id to different people?

**Comment by jianqiaozhu** [comment]
Actually this might be an incognito issue.

**Comment by jianqiaozhu** [comment]
I also just found out that even without the incognito windows, just keep spamming "New Participants" in the local debug mode, one can make earlier started participants can share the exact same participant_id with late starters. How should I handle this issue? I have a concern that many of my participants are actually one person (possibly using 1 identifiable network) behind multiple different Prolific accounts.

**Comment by pmcharrison** [comment]
I would be careful about this conclusion, I don't think it's possible for the database to create two participant rows with the same participant_id, or two participant rows with the same worker ID. If you spam "New participants" in local debug that could just be confusing the local cookie system and mixing up the participant sessions.
